accepted/00000-enable-ansible-control-node-in-registered-system.md

```diff
@@ -0,0 +1,136 @@
+
+* Once we have this, we can trigger the bootstrap of those systems by calling "salt-ssh" like this:
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i /srv/susemanager/salt/salt_ssh/mgr_ssh_id -o StrictHostKeyChecking=no -o User=root -W %h:%p uyuni-ansible-control-node.tf.local"' --roster=ansible --roster-file=/var/cache/ansible/ansible-inventory.yaml -N webserver1 state.apply certs,bootstrap pillar='{"mgr_server": "uyuni-srv.tf.local", "minion_id": "uyuni-ansible-sles15sp1-2.tf.local"}'
+```
+In esence, we apply the same states that we do at the time of Boostrapping a new minion via the UI, passing the necessary information as pillar data.
+
+* Another example of usage (without using the ansible roster/targetting):
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i PATH_TO_SALT_SSH_KEY_IN_THE_UYUNI_SERVER -o StrictHostKeyChecking=no -W %h:%p uyuni-stable-ansible-controller.tf.local /usr/bin/ssh -i PATH_TO_THE_SSH_KEY_IN_ANSIBLE_CONTROL_NODE uyuni-stable-ansible-opensuse152-2.tf.local" -o StrictHostKeyChecking=no' uyuni-stable-ansible-opensuse152-2.tf.local test.ping
+```
+
+
+### Triggering playbook executions in an Ansible control node
+
+After exploring the Ansible control node, inventory and playbooks, we can allow Uyuni to trigger the execution of one of those playbooks via a simple Salt state to execute in that control node:
+
+```yaml
+execute_ansible_playbook:
+  ansible.playbooks:
+    - name: /some/path/in/my/ansible/control/node/playbook_1.yaml
+```
+
+We would need to create a new type of Action in Uyuni for "running a playbook". We want this action to appear in the system event history for the Ansible control node.
+
+
+### Shipping Ansible for SLE15+
+
+At the time of writing this RFC, Ansible is not shipped in SLE15 any SP. In order to allow a posible Ansible control node running on SLE, we would need to ship Ansible probably in the SLE15 client tools channels.
+
+- Which version to ship? Probably Ansible 2.9, which is currently packaged for Leap, and we have it also in Factory: https://build.opensuse.org/package/show/openSUSE:Factory/ansible
+
+Considerations and other discussions about shipping Ansible in SLE, the support level, etc, can be discussed outside this RFC since it's out of the main focus of the general Ansible integration in Uyuni.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance? Bootstrapping Ansible managed clients might take some time since "salt-ssh" is involved.
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future?
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs/options have been considered? An alternative approach based on VHM/"ansible-gatherer" is discussed [in this other RFC](https://github.com/uyuni-project/uyuni-rfc/pull/53)
```

Comment:
I'd recommend trying to put down pros and cons to the extent possible. At least I would recommend making it clear that the basic premise of this design (controller is registered) could potentially be limiting compared to the other design.

Reply:
Why would it be limiting? On the contrary, I see this proposal providing a much better integration than the other one. The cases the registered Ansible controller makes more difficult are cases are do not really want to get into since they would provide a worse user experience or a more difficult to support integration.

Reply:
Perhaps that was a wrong choice of words - what I mean is "less general" - it just poses an additional requirement (registration) that the other approach does not have.

Reply:
I'd not define it a "worse or better" user experience though, it would be just different because different is the approach and the resulting number of things one can do with an onboarded/imported system.

First bunch of differences I see (@meaksh up to you to organize them in pros and cons) :
- A registered system can be managed, an imported one can just show up in the list
- A registered system needs to be of a supported flavor by us, an imported one can be of any kind
- Replacing a system would means deleting the profile and re-registering it, an imported one simply need to keep the same IP and can be swapped

On the other side I see the "limiting" word as another way of saying "stricter integration", which is fine as long as it is intended to be. I'd say the other RFC was born on a different perspective, now we explored the two paths deeply, and we see differences. Probably it was worth for the awareness sake and for the future. We can always add the "pre-step" of import systems only. I agree it would be good to keep it as a note in the current RFC here to have a full picture in one view.

Reply:
Thanks for the feedback here!

I've just added some notes in the RFC text to highlight some of the relevant different between the two approaches without really evaluating them in Pros/Cons. Just for the record, so we can even take it into consideration for future enhacements of the current integration proposed in the present RFC.